### PR TITLE
XML documentation and warning cleanup

### PR DESCRIPTION
**Pull request by brantburnett**
Motivation
----------
Generate XML documentation for inclusion in the Nuget package, while also
preparing it for Sandcastle documentation generation (LINQ-5).  Also
reduce compiler warnings.

Modifications
-------------
Updated project definition to generate XML documentation files.  Cleaned
up various issues in the XML documentation, including errors and public
members which were missing documentation.

Also, fixed several compiler warnings.  Used pragmas to disable some
obsolete warnings on the Couchbase SDK.

Results
-------
Valid XML documentation is now generated for inclusion in the Nuget
package and for later use with Sandcastle to generate HTML documentation.
There are no more compiler warnings issued.

### CODE REVIEW

File: Src/Couchbase.Linq/Extensions/EnumerableExtensions.cs [hunk 1]
```diff
@@ -9,6 +9,10 @@
 
 namespace Couchbase.Linq.Extensions
 {
+    /// <summary>
+    /// Extentions to <see cref="IEnumerable{T}"/> which emulate query extensions in <see cref="QueryExtensions"/>.
```
Comment (jeffrymorris) [inline Src/Couchbase.Linq/Extensions/EnumerableExtensions.cs]: Extentions should be Extensions.

File: Src/Couchbase.Linq/BucketContext.cs [hunk 2]
```diff
@@ -42,8 +46,8 @@ public ClientConfiguration Configuration
         }
 
         /// <summary>
-        /// Queries the current <see cref="IBucket" /> for entities of type <see cref="T" />. This is the target of
-        /// the Linq query requires that the associated JSON document have a type property that is the same as <see cref="T" />.
+        /// Queries the current <see cref="IBucket" /> for entities of type T. This is the target of
+        /// the Linq query requires that the associated JSON document have a type property that is the same as T.
```
Comment (jeffrymorris) [inline Src/Couchbase.Linq/BucketContext.cs]: should be "This is the target of a Linq query and requires that the associated JSON document have a type property that is the same as..."

File: Src/Couchbase.Linq/IBucketContext.cs [hunk 3]
```diff
@@ -40,8 +40,8 @@ public interface IBucketContext : IBucketQueryable
         void FlushChanges();
 
         /// <summary>
-        /// Queries the current <see cref="IBucket"/> for entities of type <see cref="T"/>. This is the target of
-        /// the Linq query requires that the associated JSON document have a type property that is the same as <see cref="T"/>.
+        /// Queries the current <see cref="IBucket"/> for entities of type T. This is the target of
+        /// the Linq query requires that the associated JSON document have a type property that is the same as T.
         /// </summary>
```
Comment (jeffrymorris) [inline Src/Couchbase.Linq/IBucketContext.cs]: Similar to above...needs an "and" between query and requires...I think ;)

**Comment by jeffrymorris** [comment]
@brantburnett -

Since this commit was specifically for documentation I was a little strict ;) 

Please fix and do a force push and I'll accept this PR. Feel free to disagree with my comments if i am wrong.

Thanks!

-Jeff

**Comment by brantburnett** [comment]
@jeffrymorris 

Your comments were all spot on, thanks for the careful review.  Changes are pushed.

Thanks,
Brant

**Comment by jeffrymorris** [comment]
Thanks Brant!

**Pull request merged**